pkg/storegateway/bucket.go

```diff
@@ -980,6 +1025,11 @@ func (s *BucketStore) LabelNames(ctx context.Context, req *storepb.LabelNamesReq
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
+	stats.update(func(stats *queryStats) {
+		stats.blocksQueried = len(sets)
+		stats.streamingSeriesExpandPostingsDuration += time.Since(begin)
```

Comment:
This time seems to also include the time to fetch series. I will push a commit to remove this.
```suggestion
```